crates/yttrium/src/sign/incoming.rs

```diff
@@ -149,6 +184,41 @@ pub fn handle(
                 })?;
             Ok(())
         } else if method.as_str() == Some("wc_sessionRequest") {
+            // Insert JSON-RPC history for incoming request (if not exists)
+            if let Some(request_id) = value.get("id").and_then(|id| match id {
+                Value::Number(n) => n.as_u64(),
+                Value::String(s) => s.parse::<u64>().ok(),
+                _ => None,
+            }) {
+                let exists = session_store
+                    .does_json_rpc_exist(request_id)
```

Comment:
Can we refactor this pattern to just be a single storage call? Specifically, a function that both checks if the request ID exists, and if it doesn't, write the new value? It would return a boolean (just like does_json_rpc_exist) but avoids a round trip and keeps the storage operations more use-case driven